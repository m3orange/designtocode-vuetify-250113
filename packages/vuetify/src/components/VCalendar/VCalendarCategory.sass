@import './_variables.scss'

// Theme
+theme(v-calendar-category) using ($material)
  .v-calendar-category__column,
  .v-calendar-category__column-header
    border-right: map-deep-get($material, 'calendar', 'line-color') $calendar-line-width solid

.v-calendar-category
  overflow: auto
  position: relative

  .v-calendar-category__category
    text-align: center

  .v-calendar-daily__day-container
    width: min-content
    .v-calendar-category__columns
      position: absolute
      height: 100%
      width: 100%
      top: 0

  .v-calendar-daily__day-body
    display: flex
    flex: 1
    width: 100%
    height: 100%

  .v-calendar-daily__head
    flex-direction: row
    width: min-content
    position: sticky
    top: 0
    background: white
    z-index: 2

  .v-calendar-daily_head-day
    width: auto
    position: unset

  .v-calendar-daily__intervals-head
    position: sticky !important
    left: 0
    top: 0
    background: white
    z-index: 2

  .v-calendar-daily_head-weekday
    position: sticky
    left: 50%
    width: $calendar-daily-head-weekday-size
    margin-left: cacl( -1 * ($calendar-daily-head-weekday-size/2))

  .v-calendar-daily_head-day-label
    width: $calendar-daily-head-day-label-size
    position: sticky
    left: 50%
    margin-left: cacl( -1 * ($calendar-daily-head-day-label-size/2))

  .v-calendar-daily__day
    min-width: $calendar-category-column-min-width

  .v-calendar-daily__intervals-body
    position: sticky
    left: 0
    background: white
    z-index: 1

  .v-calendar-daily__body
    overflow: unset
    .v-calendar-daily__scroll-area
      overflow-y: unset

  .v-calendar-daily__pane
    overflow-y: unset

  .v-calendar-category__columns
    display: flex
    width: min-content

    .v-calendar-category__column,
    .v-calendar-category__column-header
      flex: 1 1 auto
      width: 0
      position: relative

    .v-calendar-category__column-header
      min-width: $calendar-category-column-min-width
