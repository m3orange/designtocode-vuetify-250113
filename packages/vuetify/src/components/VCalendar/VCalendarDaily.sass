@import './_variables.scss'

// Theme
+theme(v-calendar-daily) using ($material)
  background-color: map-deep-get($material, 'calendar', 'background-color')
  border-left: map-deep-get($material, 'calendar', 'line-color') $calendar-line-width solid
  border-top: map-deep-get($material, 'calendar', 'line-color') $calendar-line-width solid

  .v-calendar-daily__intervals-head
    border-right: map-deep-get($material, 'calendar', 'line-color') $calendar-line-width solid

    &::after
      background: map-deep-get($material, 'calendar', 'line-color')
      background: linear-gradient(90deg, transparent, map-deep-get($material, 'calendar', 'line-color'))

  .v-calendar-daily_head-day
    border-right: map-deep-get($material, 'calendar', 'line-color') $calendar-line-width solid
    border-bottom: map-deep-get($material, 'calendar', 'line-color') $calendar-line-width solid
    color: map-deep-get($material, 'calendar', 'text-color')

    &.v-past
      .v-calendar-daily_head-weekday,
      .v-calendar-daily_head-day-label
        color: map-deep-get($material, 'calendar', 'past-color')

  .v-calendar-daily__intervals-body
    border-right: map-deep-get($material, 'calendar', 'line-color') $calendar-line-width solid

    .v-calendar-daily__interval-text
      color: map-deep-get($material, 'calendar', 'interval-color')

  .v-calendar-daily__day
    border-right: map-deep-get($material, 'calendar', 'line-color') $calendar-line-width solid
    border-bottom: map-deep-get($material, 'calendar', 'line-color') $calendar-line-width solid
    min-width: $calendar-category-column-min-width

  .v-calendar-daily__day-interval
    border-top: map-deep-get($material, 'calendar', 'line-color') $calendar-line-width solid

    &:first-child
      border-top: none !important

  .v-calendar-daily__interval::after
    border-top: map-deep-get($material, 'calendar', 'line-color') $calendar-line-width solid

.v-calendar-daily
  display: flex
  flex-direction: column
  height: 100%
  overflow: auto
  position: relative

.v-calendar-daily__head
  flex: none
  display: flex
  flex-direction: row
  width: min-content
  position: sticky
  top: 0
  background: white
  z-index: 2

.v-calendar-daily__intervals-head
  flex: none
  position: sticky
  left: 0
  top: 0
  background: white
  z-index: 2

  &::after
    position: absolute
    bottom: 0px
    height: $calendar-line-width
    left: 0
    right: 0
    content: ''

.v-calendar-daily_head-day
  flex: 1 1 auto

.v-calendar-daily_head-weekday
  user-select: none
  padding: $calendar-daily-weekday-padding
  font-size: $calendar-daily-weekday-font-size
  text-align: center
  text-transform: uppercase
  position: sticky
  left: 50%
  width: $calendar-daily-head-weekday-size
  margin-left: cacl( -1 * ($calendar-daily-head-weekday-size/2))

.v-calendar-daily_head-day-label
  user-select: none
  padding: $calendar-daily-day-padding
  cursor: pointer
  text-align: center
  width: $calendar-daily-head-day-label-size
  position: sticky
  left: 50%
  margin-left: cacl( -1 * ($calendar-daily-head-day-label-size/2))

.v-calendar-daily__body
  flex: 1 1 60%
  display: flex
  position: relative
  flex-direction: column

.v-calendar-daily__pane
  width: 100%
  flex: none
  display: flex
  align-items: flex-start

.v-calendar-daily__day-container
  display: flex
  flex: 1
  width: min-content
  height: 100%

.v-calendar-daily__intervals-body
  flex: none
  user-select: none
  position: sticky
  left: 0
  background: white
  z-index: 1

.v-calendar-daily__interval
  text-align: $calendar-daily-interval-gutter-align
  padding-right: $calendar-daily-interval-gutter-line-width
  border-bottom: none
  position: relative

  &::after
    width: $calendar-daily-interval-gutter-line-width
    position: absolute
    height: $calendar-line-width
    display: block
    content: ''
    right: 0
    bottom: -$calendar-line-width

.v-calendar-daily__interval-text
  display: block
  position: relative
  top: $calendar-daily-interval-gutter-top
  font-size: $calendar-daily-interval-gutter-font-size
  padding-right: $calendar-daily-interval-gutter-width

.v-calendar-daily__day
  flex: 1
  width: 0
  position: relative
