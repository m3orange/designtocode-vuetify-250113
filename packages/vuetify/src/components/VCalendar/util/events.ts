import {
  parseTimestamp,
  getDayIdentifier,
  getTimestampIdentifier,
  isTimedless,
  updateHasTime,
} from './timestamp'
import { CalendarTimestamp, CalendarEvent, CalendarEventParsed } from 'vuetify/types'

// 将原始时间解析成一个新对象，时间的年月日时分组合 成数字  例如202001011020 2020年一月一日十点二十分
export function parseEvent (
  input: CalendarEvent,
  index: number,
  startProperty: string,
  endProperty: string,
  timed = false,
  category: string | false = false,
): CalendarEventParsed {
  const startInput = input[startProperty]
  const endInput = input[endProperty]
  const startParsed: CalendarTimestamp = parseTimestamp(startInput, true)
  const endParsed: CalendarTimestamp = (endInput ? parseTimestamp(endInput, true) : startParsed)

  const start: CalendarTimestamp = isTimedless(startInput)
    ? updateHasTime(startParsed, timed) // 确定该事件与当前时间点的关系，是在前还是后
    : startParsed
  const end: CalendarTimestamp = isTimedless(endInput)
    ? updateHasTime(endParsed, timed)
    : endParsed

  const startIdentifier: number = getDayIdentifier(start)
  const startTimestampIdentifier: number = getTimestampIdentifier(start)
  const endIdentifier: number = getDayIdentifier(end)
  const endOffset: number = start.hasTime ? 0 : 2359
  const endTimestampIdentifier: number = getTimestampIdentifier(end) + endOffset
  const allDay = !start.hasTime
  return { input, start, startIdentifier, startTimestampIdentifier, end, endIdentifier, endTimestampIdentifier, allDay, index, category }
}

// 过滤该日的日历事件
export function isEventOn (event: CalendarEventParsed, dayIdentifier: number): boolean {
  // 根据日子（2022-12-03）这样的格式来判断是否相交
  // 排除法 事件的开始事件大于该日子 肯定无相交；
  //       事件的结束事件小于该日子 肯定无相交
  // 总结，事件的该日子 开始时间小于等于该日子  结束时间大于等于该日子  则肯定有段时间事件在该日子内
  return dayIdentifier >= event.startIdentifier && dayIdentifier <= event.endIdentifier
}

export function isEventHiddenOn (event: CalendarEventParsed, day: CalendarTimestamp): boolean {
  return event.end.time === '00:00' && event.end.date === day.date && event.start.date !== day.date
}

export function isEventStart (event: CalendarEventParsed, day: CalendarTimestamp, dayIdentifier: number, firstWeekday: number): boolean {
  return dayIdentifier === event.startIdentifier || (firstWeekday === day.weekday && isEventOn(event, dayIdentifier))
}
// 堆叠事件判断
export function isEventOverlapping (event: CalendarEventParsed, startIdentifier: number, endIdentifier: number): boolean {
  return startIdentifier <= event.endIdentifier && endIdentifier >= event.startIdentifier
}
