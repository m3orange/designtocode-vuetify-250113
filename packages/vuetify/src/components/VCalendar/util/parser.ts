import { CalendarCategory, CalendarCategoryTextFunction } from 'types'

export function parsedCategoryText (
  category: CalendarCategory,
  categoryText: string | CalendarCategoryTextFunction
): string {
  return typeof categoryText === 'string' && typeof category === 'object' && category
    ? category[categoryText]
    : typeof categoryText === 'function'
      ? categoryText(category)
      : category
}

export function getParsedCategories (
  categories: CalendarCategory | CalendarCategory[],
  categoryText: string | CalendarCategoryTextFunction
): CalendarCategory[] {
  if (typeof categories === 'string') return categories.split(/\s*,\s/)
  if (Array.isArray(categories)) {
    return categories.map((category: CalendarCategory) => {
      const categoryName = typeof category === 'string' ? category
        : typeof category === 'object' && category && typeof category.categoryName === 'string' ? category.categoryName
          : parsedCategoryText(category, categoryText)
      if (typeof category === 'string') return { categoryName }
      return Object.assign({ categoryName }, category)
    })
  }
  return []
}
