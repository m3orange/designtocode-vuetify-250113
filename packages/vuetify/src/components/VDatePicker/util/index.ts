import { createItemTypeNativeListeners, createItemTypeListeners } from './eventHelpers'
import createNativeLocaleFormatter from './createNativeLocaleFormatter'
import monthChange from './monthChange'
import pad from './pad'

export {
  createItemTypeNativeListeners,
  createNativeLocaleFormatter,
  createItemTypeListeners,
  monthChange,
  pad,
}
