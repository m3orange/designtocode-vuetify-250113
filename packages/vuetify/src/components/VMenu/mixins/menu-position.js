/**
 * Menu position
 *
 * @mixin
 *
 * Used for calculating an automatic position (used for VSelect)
 * Will position the VMenu content properly over the VSelect
 */
/* @vue/component */
export default {
  data: () => ({
    calculatedTopAuto: 0
  }),
  methods: {
    setScrollPosition () {
      const $el = this.$refs.content
      if (!$el || this.selectedIndex === null) return

      const activeItem = this.tiles[this.selectedIndex]
      if (activeItem) {
        $el.scrollTop = activeItem.offsetTop - $el.offsetHeight / 2 + activeItem.offsetHeight / 2
      } else {
        $el.scrollTop = 0
      }
    },
    calcLeftAuto () {
      if (this.isAttached) return 0

      return parseInt(this.dimensions.activator.left - this.defaultOffset * 2)
    },
    calcTopAuto () {
      const selectedIndex = Array.from(this.tiles)
        .findIndex(n => n.classList.contains('v-list__tile--active'))

      if (selectedIndex === -1) {
        this.selectedIndex = null

        return this.computedTop
      }

      const tileHeight = this.tiles[selectedIndex].offsetHeight

      this.selectedIndex = selectedIndex
      this.stopIndex = this.tiles.length > 4
        ? this.tiles.length - 4
        : this.tiles.length
      let additionalOffset = this.defaultOffset
      let offsetPadding

      // Menu should be centered
      if (selectedIndex > this.startIndex &&
        selectedIndex < this.stopIndex
      ) {
        offsetPadding = 1.5 * tileHeight
      // Menu should be offset top
      } else if (selectedIndex >= this.stopIndex) {
        // Being offset top means
        // we have to account for top
        // and bottom list padding
        additionalOffset *= 2
        offsetPadding = (selectedIndex - this.stopIndex) * tileHeight
      // Menu should be offset bottom
      } else {
        offsetPadding = selectedIndex * tileHeight
      }

      return (
        this.computedTop +
        additionalOffset -
        offsetPadding -
        (tileHeight / 2)
      )
    }
  }
}
