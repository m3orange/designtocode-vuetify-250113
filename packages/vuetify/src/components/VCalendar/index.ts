// Styles
import './VCalendarCategory.sass'
import './VCalendarDaily.sass'
import './VCalendarWeekly.sass'

import VCalendar from './VCalendar'
import VCalendarDaily from './VCalendarDaily'
import VCalendarWeekly from './VCalendarWeekly'
import VCalendarMonthly from './VCalendarMonthly'
import VCalendarCategory from './VCalendarCategory'

export { VCalendar, VCalendarCategory, VCalendarDaily, VCalendarWeekly, VCalendarMonthly }

export default {
  $_vuetify_subcomponents: {
    VCalendar,
    VCalendarCategory,
    VCalendarDaily,
    VCalendarWeekly,
    VCalendarMonthly,
  },
}
