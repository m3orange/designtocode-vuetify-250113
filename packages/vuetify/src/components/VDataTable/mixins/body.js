import ExpandTransitionGenerator from '../../transitions/expand-transition'

import { getObjectValueByPath } from '../../../util/helpers'

/* @vue/component */
export default {
  methods: {
    genTBody () {
      const children = this.genItems()

      return this.$createElement('tbody', children)
    },
    genExpandedRow (props) {
      const children = []

      if (this.isExpanded(props.item)) {
        const expand = this.$createElement('div', {
          class: 'v-datatable__expand-content',
          key: getObjectValueByPath(props.item, this.itemKey)
        }, [this.$scopedSlots.expand(props)])

        children.push(expand)
      }

      const transition = this.$createElement('transition-group', {
        class: 'v-datatable__expand-col',
        attrs: { colspan: this.headerColumns },
        props: {
          tag: 'td'
        },
        on: ExpandTransitionGenerator('v-datatable__expand-col--expanded')
      }, children)

      return this.genTR([transition], { class: 'v-datatable__expand-row' })
    },
    getItemKeyOrDefault (props, defalutVal) {
      return this.itemKey ? getObjectValueByPath(props.item, this.itemKey) : defalutVal
    },
    genFilteredItems () {
      if (!this.$scopedSlots.items) {
        return null
      }

      const rows = []
      for (let index = 0, len = this.filteredItems.length; index < len; ++index) {
        const item = this.filteredItems[index]
        const props = this.createProps(item, index)
        let row = this.$scopedSlots.items(props)

        if (this.hasTag(row, 'td')) {
          row = this.genTR(row, {
            key: this.getItemKeyOrDefault(props, index),
            attrs: { active: this.isSelected(item) }
          })
        } else if (row.length === 1 && !row[0].key) {
          row[0].key = this.getItemKeyOrDefault(props, index)
        } else {
          // reach cases of bad usage
        }
        rows.push(row)
        if (this.$scopedSlots.expand) {
          const expandRow = this.genExpandedRow(props)
          rows.push(expandRow)
        }
      }

      return rows
    },
    genEmptyItems (content) {
      if (this.hasTag(content, 'tr')) {
        return content
      } else if (this.hasTag(content, 'td')) {
        return this.genTR(content)
      } else {
        return this.genTR([this.$createElement('td', {
          class: {
            'text-xs-center': typeof content === 'string'
          },
          attrs: { colspan: this.headerColumns }
        }, content)])
      }
    }
  }
}
