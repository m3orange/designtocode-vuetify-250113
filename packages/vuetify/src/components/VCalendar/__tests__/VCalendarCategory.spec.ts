import {
  mount,
  Wrapper,
  MountOptions,
} from '@vue/test-utils'
import { ExtractVue } from '../../../util/mixins'
import VCalendar from '../VCalendar'

describe('VCalendarCategory', () => {
  type Instance = ExtractVue<typeof VCalendar>
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(VCalendar, {
        ...options,
        mocks: {
          $vuetify: {
            lang: {
              current: 'en-US',
            },
          },
        },
      })
    }
  })

  it('should test categoryText prop as a string', () => {
    const wrapper = mountFunction({
      propsData: {
        type: 'category',
        events: [{
          name: 'TestEvent',
          start: new Date(),
          end: new Date(),
          category: 'Nate',
        }],
        categories: [{ name: 'Nate' }],
        categoryText: 'name',
      },
    })

    expect(wrapper.find('.v-calendar-category__column-header').text()).toEqual('Nate')
  })

  it('should test categoryText prop as a function', () => {
    const wrapper = mountFunction({
      propsData: {
        type: 'category',
        events: [{
          name: 'TestEvent',
          start: new Date(),
          end: new Date(),
          category: '20',
        }],
        categories: [{ name: 'Nate', age: '20' }],
        categoryText (category) {
          return category.age
        },
      },
    })

    expect(wrapper.find('.v-calendar-category__column-header').text()).toEqual('20')
  })

  it('should pass entire cateogry to interval style method', () => {
    function intervalStyle (obj) {
      expect(obj.category).toEqual({ name: 'Nate', age: 20, categoryName: 'Nate' })
    }

    const wrapper = mountFunction({
      propsData: {
        type: 'category',
        categories: [{ name: 'Nate', age: 20 }],
        categoryText: 'name',
        events: [{
          name: 'TestEvent',
          start: new Date(),
          end: new Date(),
          category: '20',
        }],
        intervalStyle,
      },
    })
  })
})
