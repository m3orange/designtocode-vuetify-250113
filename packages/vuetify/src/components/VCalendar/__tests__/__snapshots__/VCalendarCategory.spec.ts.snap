// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VCalendarCategory should render correctly without shortIntervals prop 1`] = `
<div class="v-calendar-daily v-calendar-category theme--light">
  <div class="v-calendar-daily__head"
       style="margin-right: 0px;"
  >
    <div class="v-calendar-daily__intervals-head"
         style="width: 60px;"
    >
    </div>
    <div class="v-calendar-daily_head-day v-past">
      <div class="v-calendar-daily_head-weekday">
        Tue
      </div>
      <div class="v-calendar-daily_head-day-label">
        <button type="button"
                class="v-btn v-btn--depressed v-btn--fab v-btn--round theme--light v-size--default transparent"
        >
          <span class="v-btn__content">
            29
          </span>
        </button>
      </div>
      <div class="v-calendar-category__columns">
        <div class="v-calendar-category__column-header">
          <div class="v-calendar-category__category">
          </div>
        </div>
      </div>
    </div>
    <div class="v-calendar-daily_head-day v-past">
      <div class="v-calendar-daily_head-weekday">
        Wed
      </div>
      <div class="v-calendar-daily_head-day-label">
        <button type="button"
                class="v-btn v-btn--depressed v-btn--fab v-btn--round theme--light v-size--default transparent"
        >
          <span class="v-btn__content">
            30
          </span>
        </button>
      </div>
      <div class="v-calendar-category__columns">
        <div class="v-calendar-category__column-header">
          <div class="v-calendar-category__category">
          </div>
        </div>
      </div>
    </div>
    <div class="v-calendar-daily_head-day v-past">
      <div class="v-calendar-daily_head-weekday">
        Thu
      </div>
      <div class="v-calendar-daily_head-day-label">
        <button type="button"
                class="v-btn v-btn--depressed v-btn--fab v-btn--round theme--light v-size--default transparent"
        >
          <span class="v-btn__content">
            31
          </span>
        </button>
      </div>
      <div class="v-calendar-category__columns">
        <div class="v-calendar-category__column-header">
          <div class="v-calendar-category__category">
          </div>
        </div>
      </div>
    </div>
    <div class="v-calendar-daily_head-day v-past">
      <div class="v-calendar-daily_head-weekday">
        Fri
      </div>
      <div class="v-calendar-daily_head-day-label">
        <button type="button"
                class="v-btn v-btn--depressed v-btn--fab v-btn--round theme--light v-size--default transparent"
        >
          <span class="v-btn__content">
            1
          </span>
        </button>
      </div>
      <div class="v-calendar-category__columns">
        <div class="v-calendar-category__column-header">
          <div class="v-calendar-category__category">
          </div>
        </div>
      </div>
    </div>
    <div class="v-calendar-daily_head-day v-past">
      <div class="v-calendar-daily_head-weekday">
        Sat
      </div>
      <div class="v-calendar-daily_head-day-label">
        <button type="button"
                class="v-btn v-btn--depressed v-btn--fab v-btn--round theme--light v-size--default transparent"
        >
          <span class="v-btn__content">
            2
          </span>
        </button>
      </div>
      <div class="v-calendar-category__columns">
        <div class="v-calendar-category__column-header">
          <div class="v-calendar-category__category">
          </div>
        </div>
      </div>
    </div>
    <div class="v-calendar-daily_head-day v-past">
      <div class="v-calendar-daily_head-weekday">
        Sun
      </div>
      <div class="v-calendar-daily_head-day-label">
        <button type="button"
                class="v-btn v-btn--depressed v-btn--fab v-btn--round theme--light v-size--default transparent"
        >
          <span class="v-btn__content">
            3
          </span>
        </button>
      </div>
      <div class="v-calendar-category__columns">
        <div class="v-calendar-category__column-header">
          <div class="v-calendar-category__category">
          </div>
        </div>
      </div>
    </div>
    <div class="v-calendar-daily_head-day v-past">
      <div class="v-calendar-daily_head-weekday">
        Mon
      </div>
      <div class="v-calendar-daily_head-day-label">
        <button type="button"
                class="v-btn v-btn--depressed v-btn--fab v-btn--round theme--light v-size--default transparent"
        >
          <span class="v-btn__content">
            4
          </span>
        </button>
      </div>
      <div class="v-calendar-category__columns">
        <div class="v-calendar-category__column-header">
          <div class="v-calendar-category__category">
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="v-calendar-daily__body">
    <div class="v-calendar-daily__scroll-area">
      <div class="v-calendar-daily__pane"
           style="height: 1152px;"
      >
        <div class="v-calendar-daily__day-container">
          <div class="v-calendar-daily__intervals-body"
               style="width: 60px;"
          >
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                01:00 AM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                02:00 AM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                03:00 AM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                04:00 AM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                05:00 AM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                06:00 AM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                07:00 AM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                08:00 AM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                09:00 AM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                10:00 AM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                11:00 AM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                12:00 PM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                01:00 PM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                02:00 PM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                03:00 PM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                04:00 PM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                05:00 PM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                06:00 PM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                07:00 PM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                08:00 PM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                09:00 PM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                10:00 PM
              </div>
            </div>
            <div class="v-calendar-daily__interval"
                 style="height: 48px;"
            >
              <div class="v-calendar-daily__interval-text">
                11:00 PM
              </div>
            </div>
          </div>
          <div class="v-calendar-daily__day v-past">
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-daily__day-interval"
                 style="height: 48px;"
            >
            </div>
            <div class="v-calendar-category__columns">
              <div class="v-calendar-category__column">
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
